gobblin-data-management/src/main/java/gobblin/data/management/version/finder/DatePartitionHiveVersionFinder.java

```diff
@@ -0,0 +1,100 @@
+ * a date pattern as per prop value {@value #PARTITION_VALUE_DATE_TIME_PATTERN_KEY}.
+ * </p>
+ * <p>
+ * E.g if the hive partition is datepartition=2016-01-10-22/field1=f1Value.
+ * The {@value #PARTITION_KEY_NAME_KEY}=datepartiton and {@value #PARTITION_VALUE_DATE_TIME_PATTERN_KEY}=yyyy-MM-dd-HH
+ *
+ * </p>
+ */
+@Slf4j
+public class DatePartitionHiveVersionFinder extends AbstractHiveDatasetVersionFinder {
+
+  public static final String PARTITION_VALUE_DATE_TIME_PATTERN_KEY = "hive.partition.value.datetime.pattern";
+  public static final String DEFAULT_PARTITION_VALUE_DATE_TIME_PATTERN = "yyyy-MM-dd-HH";
+
+  public static final String PARTITION_VALUE_DATE_TIME_TIMEZONE_KEY = "hive.partition.value.datetime.timezone";
+  public static final String DEFAULT_PARTITION_VALUE_DATE_TIME_TIMEZONE = ConfigurationKeys.PST_TIMEZONE_NAME;
+
+  public static final String PARTITION_KEY_NAME_KEY = "hive.partition.key.name";
+  public static final String DEFAULT_PARTITION_KEY_NAME = "datepartition";
+
+  protected final DateTimeFormatter formatter;
+  private final String partitionValueDatePattern;
+  private final String partitionKeyName;
+
+  public DatePartitionHiveVersionFinder(FileSystem fs, Config config) {
+
+    String pattern = ConfigUtils.getString(config, PARTITION_VALUE_DATE_TIME_PATTERN_KEY, DEFAULT_PARTITION_VALUE_DATE_TIME_PATTERN);
+
+    if (config.hasPath(PARTITION_VALUE_DATE_TIME_TIMEZONE_KEY)) {
+      this.formatter = DateTimeFormat.forPattern(pattern).withZone(DateTimeZone.forID(config.getString(PARTITION_VALUE_DATE_TIME_TIMEZONE_KEY)));
+    } else {
+      this.formatter = DateTimeFormat.forPattern(pattern).withZone(DateTimeZone.forID(DEFAULT_PARTITION_VALUE_DATE_TIME_TIMEZONE));
+    }
+
+    this.partitionValueDatePattern = pattern;
+    this.partitionKeyName = ConfigUtils.getString(config, PARTITION_KEY_NAME_KEY, DEFAULT_PARTITION_KEY_NAME);
+
+  }
+
+  /**
+   * Create a {@link TimestampedHiveDatasetVersion} from a {@link Partition}. The hive table is expected
+   * to be date partitioned by {@link #partitionKeyName}. The partition value format must be {@link #partitionValueDatePattern}
+   * <p>
+   * The method throws an exception when the date pattern for the partition name is invalid
+   * </p>
+   * {@inheritDoc}
+   * @see gobblin.data.management.version.finder.AbstractHiveDatasetVersionFinder#getDatasetVersion(org.apache.hadoop.hive.ql.metadata.Partition)
+   */
+  @Override
+  protected TimestampedHiveDatasetVersion getDatasetVersion(Partition partition) {
+
+    /*
+     * The datetime string is extracted from the partition name. Partition name combines all the partition fields of the table.
+     * E.g it is of the form "datepartition=2016-01-10-22/field1=f1Value". From this name we need to extract 2016-01-10-22.
+     */
+    try {
+      String dateTimeString =
+          StringUtils.substring(StringUtils.substringAfter(partition.getName(), partitionKeyName + "="), 0, this.partitionValueDatePattern.length());
```

Comment:
You can use `partition.getValues()` instead. It contains the values for the partition columns.